cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -501,6 +501,14 @@ func (csr *ClusterStateRegistry) IsNodeGroupScalingUp(nodeGroupName string) bool
 	return found
 }
 
+// HasNodeGroupStartedScaleUp returns true if the node group has started scale up.
+// Difference between IsNodeGroupScalingUp() is, this function doesn't check whether
+// there are any upcoming nodes for the node group.
+func (csr *ClusterStateRegistry) HasNodeGroupStartedScaleUp(nodeGroupName string) bool {
```

Comment:
Why is this not mutex protected? FWIW, looks like `IsNodeGroupScalingUp` should really be `isNodeGroupScalingUpNoLock` since it is used internally from mutex-protected public functions.

Reply:
Added mutex protection, however left renaming the `IsNodeGroupScalingUp` to another pr.